spec/services/record_declarations/ecf/early_career_teacher_spec.rb

```diff
@@ -3,7 +3,8 @@
 require "rails_helper"
 
 RSpec.describe RecordDeclarations::ECF::EarlyCareerTeacher do
-  let(:cpd_lead_provider) { create(:cpd_lead_provider) }
+  let(:ecf_lead_provider) { create(:lead_provider) }
+  let(:cpd_lead_provider) { create(:cpd_lead_provider, lead_provider: ecf_lead_provider) }
```

Comment:
Incidentally, I want to make the `[ecf_]lead_provider` factory automatically generate its parent `cpd_lead_provider` record, but I'd already spent far too long on the factories and this patch was the first variant where I got everything to work together. Further iteration welcome!